osc-server/src/main/java/org/osc/core/broker/rest/client/k8s/KubernetesDeploymentApi.java

```diff
@@ -0,0 +1,210 @@
+/*******************************************************************************
+ * Copyright (c) Intel Corporation
+ * Copyright (c) 2017
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package org.osc.core.broker.rest.client.k8s;
+
+import org.apache.log4j.Logger;
+import org.osc.core.broker.service.exceptions.VmidcException;
+
+import io.fabric8.kubernetes.api.model.extensions.Deployment;
+import io.fabric8.kubernetes.api.model.extensions.DeploymentBuilder;
+import io.fabric8.kubernetes.client.KubernetesClientException;
+
+/**
+ * This class implements the pod related operations used by other
+ * osc-core components.
+ */
+public class KubernetesDeploymentApi extends KubernetesApi {
+    private final static String OSC_DEPLOYMENT_LABEL_NAME = "osc-deployment";
+
+    private static final Logger LOG = Logger.getLogger(KubernetesDeploymentApi.class);
+
+    public KubernetesDeploymentApi(KubernetesClient client) {
+        super(client.getClient());
+    }
+
+    /**
+     * Retrieves the deployment with the given uid, namespace and name
+     * <p>
+     * The additional parameters namespace and name are needed because K8s APIs
+     * do not support queries by uid
+     *
+     * @param uid  the unique identifier of the deployment to be retrieved
+     * @param namespace  the namespace of the deployment to be retrieved
+     * @param name  the name of the deployment to be retrieved
+     * @return the deployment with the given namespace, name and uid. NULL if no deployment is found
+     * @throws VmidcException  if a K8s SDK specific exception is caught
+     */
+    public KubernetesDeployment getDeploymentById(String uid, String namespace, String name) throws VmidcException {
+        if (uid == null) {
+            throw new IllegalArgumentException("Uid should not be null");
+        }
+
+        if (name == null) {
+            throw new IllegalArgumentException("Name should not be null");
+        }
+
+        if (namespace == null) {
+            throw new IllegalArgumentException("Namespace should not be null");
+        }
+
+        KubernetesDeployment deployment = getDeploymentByName(namespace, name);
```

Comment:
So, namespace and name uniquely identify it, but we also need to know the id?

Reply:
Correct, the name and namespace are unique in space, not in time. In other words, if an entity is deleted from k8s with a given name and namespace another can be recreated with the same namespace/name however the id will be different. We want to make sure we are referring to the correct entity also in time.